app/assets/javascripts/components/signup.js

```diff
@@ -0,0 +1,49 @@
+$( document ).ready(function() {
+
+  $('.error').hide();
+
+  $(".signupForm").submit(function(event) {
+
+  var errorMessage = "";
+
+  event.preventDefault();
+
+  function isValidFullName(fullName) {
+    var validate_name = new RegExp (/^[a-zA-Z ]+$/);
+    return validate_name.test(fullName);
+  }
+
+  function isValidPassword(password) {
+    var validate_password = new RegExp (/^[^\s]+$/);
+    return validate_password.test(password);
+  }
+
+  function isValidEmailAddress(emailAddress) {
+
+    var pattern = new RegExp(/^((([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*)|((\x22)((((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(([\x01-\x08\x0b\x0c\x0e-\x1f\x7f]|\x21|[\x23-\x5b]|[\x5d-\x7e]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(\\([\x01-\x09\x0b\x0c\x0d-\x7f]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))))*(((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(\x22)))@((([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.?$/i);
```

Comment:
This is the official standard for what an email should be but it only checks for basic syntax.  So an email address like `toan@wontons.are.awesome` would get past this filter even though it's obviously invalid.  Also, this RegEx allows `""` and `[]` which can break some email clients/scripts.

My preferred RegEx for email validation is this:
`[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@
(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+(?:[A-Z]{2}|com|org|net|edu|gov|mil|
biz|info|mobi|name|aero|asia|jobs|museum)\b`

This removes quotes and square brackets and limits the TLD to any 2-letter country code or those specified in the regex.  This isn't the most future-proof regex but it'll work for almost all emails used today.